### PR TITLE
One more attempt to fix retries in clickhouse-test

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by alesapin** [comment]
image
???

**Comment by alesapin** [comment]
Let's try one more time. `Connection loss`/`Session expired` errors look fixed, but tests in DatabaseReplicated fail for other reasons.

**Comment by alesapin** [comment]
Only `Functional stateless tests (release, DatabaseReplicated)` is important

**Comment by alesapin** [comment]
`01622_defaults_for_url_engine` -- absolutely random test. I'll merge.

**Comment by alesapin** [comment]
DecomposeFloat -- reported https://github.com/ClickHouse/ClickHouse/issues/24283

File: tests/clickhouse-test [hunk 1]
```diff
@@ -183,15 +184,17 @@ def run_single_test(args, ext, server_logs_level, client_options, case_file, std
         'stderr': stderr_file,
     }
 
-    pattern = '{test} > {stdout} 2> {stderr}'
+    # >> append to stdout and stderr, because there are also output of per test database creation
+    if not args.database:
+        pattern = '{test} >> {stdout} 2>> {stderr}'
```
Comment (azat) [inline tests/clickhouse-test:189]: Now stdout never truncated and multiple test runs will duplicate the stdout reference, and stdout is not used in `CREATE DATABASE` query, so it should be overwritten, fixed in https://github.com/ClickHouse/ClickHouse/pull/24411/commits/2129335147dcd6f0092b7488bf2f5d4717c1fcc6

**Pull request merged**